govcd/nsxt_edge_cluster.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// NsxtEdgeCluster is a logical grouping of NSX-T Edge virtual machines.
+type NsxtEdgeCluster struct {
+	NsxtEdgeCluster *types.NsxtEdgeCluster
+	client          *Client
+}
+
+// GetNsxtEdgeClusterByName retrieves a particular NSX-T Edge Cluster by name available for that VDC
+func (vdc *Vdc) GetNsxtEdgeClusterByName(name string) (*NsxtEdgeCluster, error) {
+	if name == "" {
+		return nil, fmt.Errorf("empty NSX-T Edge Cluster name specified")
+	}
+
+	// Ideally FIQL filter could be used to filter on server side and get only desired result, but filtering on
+	// 'name' is not yet supported. The only supported field for filtering is
+	// _context==urn:vcloud:vdc:09722307-aee0-4623-af95-7f8e577c9ebc to specify parent Org Vdc (This
+	// automatically happens in GetAllNsxtEdgeClusters()). The below filter injection is left as documentation.
+	/*
+		queryParameters := copyOrNewUrlValues(nil)
+		queryParameters.Add("filter", "name=="+name)
+	*/
+
+	nsxtEdgeClusters, err := vdc.GetAllNsxtEdgeClusters(nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not find NSX-T Edge Cluster with name '%s' for Org Vdc with id '%s': %s",
+			name, vdc.Vdc.ID, err)
+	}
+
+	// TODO remove this when FIQL supports filtering on 'name'
+	nsxtEdgeClusters = filterNsxtEdgeClusters(name, nsxtEdgeClusters)
+	// EOF TODO remove this when FIQL supports filtering on 'name'
+
+	if len(nsxtEdgeClusters) == 0 {
+		// ErrorEntityNotFound is injected here for the ability to validate problem using ContainsNotFound()
+		return nil, fmt.Errorf("%s: no NSX-T Tier-0 Edge Cluster with name '%s' for Org Vdc with id '%s' found",
+			ErrorEntityNotFound, name, vdc.Vdc.ID)
+	}
+
+	if len(nsxtEdgeClusters) > 1 {
+		return nil, fmt.Errorf("more than one (%d) NSX-T Edge Cluster with name '%s' for Org Vdc with id '%s' found",
+			len(nsxtEdgeClusters), name, vdc.Vdc.ID)
+	}
+
+	return nsxtEdgeClusters[0], nil
+}
+
+func filterNsxtEdgeClusters(name string, allNnsxtEdgeCluster []*NsxtEdgeCluster) []*NsxtEdgeCluster {
+	filteredNsxtEdgeClusters := make([]*NsxtEdgeCluster, 0)
```

Comment:
Is this means that you create clusters with same name? - maybe worth to add to function docs :)

Reply:
Edge Clusters can have the same name, but the idea is still that one needs to "find" by name. Problem is that API filtering does not support it.
I have added extra note to the `GetNsxtEdgeClusterByName` function about it